int32 leftPulse
int32 rightPulse
